AlarmPlus/Views/AlarmSettingsViewController.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  AlarmSettingsTableViewController.swift
+//  AlarmPlus
+//
+//  Created by Charlie Scheer on 11/29/18.
+//  Copyright © 2018 Praxsis. All rights reserved.
+//
+
+import UIKit
+
+class AlarmSettingsViewController: UIViewController {
+    var alarm : Alarm?
+    var alarmCurrentDays : [Int] = []
+    var daySelectButtonsArray : [UIButton] = []
+    
+    @IBOutlet weak var datePicker: UIDatePicker!
+    @IBOutlet weak var sundayButton: UIButton!
+    @IBOutlet weak var mondayButton: UIButton!
+    @IBOutlet weak var tuesdayButton: UIButton!
+    @IBOutlet weak var wednesdayButton: UIButton!
+    @IBOutlet weak var thursdayButton: UIButton!
+    @IBOutlet weak var fridayButton: UIButton!
+    @IBOutlet weak var saturdayButton: UIButton!
+    
+    
+    @IBAction func dateWasPressed(_ sender: Any) {
+        guard let buttonPressed = sender as? UIButton else {
+            return
+        }
+        if buttonPressed.isSelected {
+            //The day that was selected is currently active in the alarm
+            
+            buttonPressed.isSelected = false
+            if alarmCurrentDays.contains(buttonPressed.tag) {
+                let fAlarmCurrentDays = alarmCurrentDays.filter {$0 != buttonPressed.tag}
+                alarmCurrentDays = fAlarmCurrentDays
+            }
+        } else {
+            //The day is pressed and currently is not active
+            
+            buttonPressed.isSelected = true
+            if alarmCurrentDays.contains(buttonPressed.tag) == false {
+                alarmCurrentDays.append(buttonPressed.tag)
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        
+        if alarm == nil {
+            setupNewAlarmInfo()
+        } else {
+            setupAlarmInfo()
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        
+    }
+    
+    
+    func setupView() {
+        let saveButton = UIBarButtonItem.init(title: "Save", style: .plain, target: self, action: #selector(rightButtonAction(sender:)))
+        navigationItem.rightBarButtonItem = saveButton
+        daySelectButtonsArray = [sundayButton, mondayButton, tuesdayButton, wednesdayButton, thursdayButton, fridayButton, saturdayButton, sundayButton]
+    }
+    
+    @objc func rightButtonAction(sender: UIBarButtonItem) {
+        //Save UIBarbutton item actions
+        var alarmsArray = MemoryFunctions.getSavedAlarmsArray()
+        
+        guard let currentAlarm = alarm else {
+            return
+        }
+        
+        let (setHour, setMinute) = getUserSetTime(datePicker: datePicker)
```

Comment:
this could be `let timeSet = getUserSetTime(datePicker: datePicker)`

and then below to use it, you would do:

`currentAlarm.schedule.setAlarmTime(hour: timeSet.hour, minute: timeSet.minute)`